plugins/catalog-backend/src/catalog/DatabaseLocationsCatalog.ts

```diff
@@ -16,11 +16,24 @@
 
 import { Database } from '../database';
 import { AddLocation, Location, LocationsCatalog } from './types';
+import { LocationReaders } from '../ingestion';
 
 export class DatabaseLocationsCatalog implements LocationsCatalog {
   constructor(private readonly database: Database) {}
 
   async addLocation(location: AddLocation): Promise<Location> {
+    const outputs = await LocationReaders.create().read(
```

Comment:
the constructor should instead take a `private readonly reader: LocationReader`, and then that same reader instance will be used in the refresh loop as well etc

Reply:
this also makes the test easier to write - you just give it a fake reader that the test itself can create

Reply:
Makes a lot of sense. I'll fix it.